### PR TITLE
[buildsystem]: Fix hiredis package version: 0.14.1-1

### PR DESCRIPTION
**Pull request by nazariig**
**Build error:**
```
18:20:27  [ finished ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb ]
18:20:27  [ FAIL LOG START ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb ]
18:20:27  [ REASON ] :      target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb does not exist  
18:20:27  [ FLAGS  FILE    ] : [mellanox amd64 buster y buster]
18:20:27  [ FLAGS  DEPENDS ] : [mellanox amd64 buster y buster]
18:20:27  [ FLAGS  DIFF    ] : []
18:20:27  make[1]: warning: -j30 forced in submake: resetting jobserver mode.
18:20:27  make[1]: Entering directory '/sonic/src/hiredis'
18:20:27  rm -rf hiredis-0.14.0
18:20:27  wget -O hiredis_0.14.0.orig.tar.gz http://http.debian.net/debian/pool/main/h/hiredis/hiredis_0.14.0.orig.tar.gz
18:20:27  wget -O hiredis_0.14.0-3~bpo9+1.debian.tar.xz http://http.debian.net/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.debian.tar.xz
18:20:27  wget -O hiredis_0.14.0-3~bpo9+1.dsc http://http.debian.net/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.dsc
18:20:27  dpkg-source -x hiredis_0.14.0-3~bpo9+1.dsc
18:20:27  pushd hiredis-0.14.0
18:20:27  dpkg-buildpackage -rfake******** -d -b -us -uc -j30 --********dir /sonic/dpkg/tmp.ALm1Mqoiea
18:20:27  popd
18:20:27  mv libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb libhiredis0.14-dbgsym_0.14.0-3~bpo9+1_amd64.deb libhiredis-dev_0.14.0-3~bpo9+1_amd64.deb /sonic/target/debs/buster/
18:20:27  /usr/local/share/buildinfo/scripts/buildinfo_base.sh: line 165: /usr/local/share/buildinfo/build-versions/versions-web: Permission denied
18:20:27  --2023-06-14 15:20:25--  http://http.debian.net/debian/pool/main/h/hiredis/hiredis_0.14.0.orig.tar.gz
18:20:27  Resolving http.debian.net (http.debian.net)... 130.89.148.77, 128.31.0.62, 149.20.4.15, ...
18:20:27  Connecting to http.debian.net (http.debian.net)|130.89.148.77|:80... connected.
18:20:27  HTTP request sent, awaiting response... 302 Found
18:20:27  Location: http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0.orig.tar.gz [following]
18:20:27  --2023-06-14 15:20:25--  http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0.orig.tar.gz
18:20:27  Resolving cdn-fastly.deb.debian.org (cdn-fastly.deb.debian.org)... 146.75.122.132, 2a04:4e42:8e::644
18:20:27  Connecting to cdn-fastly.deb.debian.org (cdn-fastly.deb.debian.org)|146.75.122.132|:80... connected.
18:20:27  HTTP request sent, awaiting response... 200 OK
18:20:27  Length: 63061 (62K) [application/x-gzip]
18:20:27  Saving to: 'hiredis_0.14.0.orig.tar.gz'
18:20:27  
18:20:27       0K .......... .......... .......... .......... .......... 81%  716K 0s
18:20:27      50K .......... .                                          100% 1.79M=0.08s
18:20:27  
18:20:27  2023-06-14 15:20:26 (809 KB/s) - 'hiredis_0.14.0.orig.tar.gz' saved [63061/63061]
18:20:27  
18:20:27  /usr/local/share/buildinfo/scripts/buildinfo_base.sh: line 165: /usr/local/share/buildinfo/build-versions/versions-web: Permission denied
18:20:27  --2023-06-14 15:20:26--  http://http.debian.net/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.debian.tar.xz
18:20:27  Resolving http.debian.net (http.debian.net)... 149.20.4.15, 130.89.148.77, 128.31.0.62, ...
18:20:27  Connecting to http.debian.net (http.debian.net)|149.20.4.15|:80... connected.
18:20:27  HTTP request sent, awaiting response... 302 Found
18:20:27  Location: http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.debian.tar.xz [following]
18:20:27  --2023-06-14 15:20:27--  http://cdn-fastly.deb.debian.org/debian/pool/main/h/hiredis/hiredis_0.14.0-3~bpo9+1.debian.tar.xz
18:20:27  Resolving cdn-fastly.deb.debian.org (cdn-fastly.deb.debian.org)... 146.75.122.132, 2a04:4e42:8e::644
18:20:27  Connecting to cdn-fastly.deb.debian.org (cdn-fastly.deb.debian.org)|146.75.122.132|:80... connected.
18:20:27  HTTP request sent, awaiting response... 404 Not Found
18:20:27  2023-06-14 15:20:27 ERROR 404: Not Found.
18:20:27  
18:20:27  make[1]: *** [Makefile:9: /sonic/target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb] Error 8
18:20:27  make[1]: Leaving directory '/sonic/src/hiredis'
18:20:27  [  FAIL LOG END  ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb ]
18:20:27  make: *** [slave.mk:658: target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb] Error 1
18:20:27  make: *** Waiting for unfinished jobs....
```

#### Why I did it
* To fix `hiredis` compilation

##### Work item tracking
* N/A

#### How I did it
* Changed package version: `0.14.0-3~bpo9+1` -> `0.14.1-1`

#### How to verify it

1. make configure PLATFORM=mellanox
2. make target/sonic-mellanox.bin

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [x] master 

#### Description for the changelog

* N/A

#### Link to config_db schema for YANG module changes

* N/A

#### A picture of a cute animal (not mandatory but encouraged)
```
      .---.        .-----------
     /     \  __  /    ------
    / /     \(  )/    -----
   //////   ' \/ `   ---
  //// / // :    : ---
 // /   /  /`    '--
//          //..\\
       ====UU====UU====
           '//||\\`
             ''``
```

### CODE REVIEW

**Comment by liushilongbuaa** [comment]
Yes, hiredis_0.14.0-3~bpo9+1 is not available on debian.org.
But it seems that  hiredis_0.14.0-3 will also deprecated at 2023-11-04.

**Comment by liushilongbuaa** [comment]
@xumia, @qiluo-msft , there are a lot of debian packages deprecated by debian.org each year.
Updating these packages to newer version is a heavy job. Can we use apt-get download instead of wget/dget?
```
apt-get download libhiredis0.14
```

**Comment by nazariig** [comment]
@liushilongbuaa this is a blocker for us. Can we temporary use this solution until long term discussion is in progress?

**Comment by liushilongbuaa** [comment]
> @liushilongbuaa this is a blocker for us. Can we temporary use this solution until long term discussion is in progress?

Currently, LGTM. But I need to check pipeline log to confirm.

File: rules/hiredis.mk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # libhiredis package
 
 HIREDIS_VERSION = 0.14.0
```
Comment (xumia) [inline rules/hiredis.mk:3]: The version 0.14-0 should be a version introduced in the old Debian release. For bullseye, can we use version 0.14.1-1?
```
root@b0889ce3a4c4:/# cat /etc/os-release | grep VERSION_CODENAME
VERSION_CODENAME=bullseye
root@b0889ce3a4c4:/# apt-cache madison libhiredis0.14
libhiredis0.14 |   0.14.1-1 | http://deb.debian.org/debian bullseye/main amd64 Packages
root@b0889ce3a4c4:/# 

```

Comment (nazariig) [reply rules/hiredis.mk]: @xumia fixed

**Comment by nazariig** [comment]
@StormLiangMS can you please merge?

**Comment by nazariig** [comment]
@xumia can you please merge if no more comments?

**Review by xumia — APPROVED** [review]

**Comment by abdosi** [comment]
@yxieca we need this for 202205 as build are failing for that also.

@gechiang for viz.

**Comment by yxieca** [comment]
@abdosi  can you point me to the build failure log? I see 202205 PR builds are passing.

**Comment by StormLiangMS** [comment]
@xumia I didn't see this failure on 202305, do we need this fix?

**Comment by saiarcot895** [comment]
This is seen on 202012 build as well.

**Comment by xumia** [comment]
> @xumia I didn't see this failure on 202305, do we need this fix?
Yes, we need it, the package is very old, has been removed in Debian official mirror.
The 20205 build not failed, because the package is still available in our mirror.

**Comment by MalteJ** [comment]
it has already been backported to 202305 via #15569

**Comment by developfast** [comment]
202205 cherry-pick: https://github.com/sonic-net/sonic-buildimage/pull/15651

**Comment by StormLiangMS** [comment]
@nazariig could you help to create separate PR for 202305, cherry pick failure.

**Comment by nazariig** [comment]
> @nazariig could you help to create separate PR for 202305, cherry pick failure.

It is already done: https://github.com/sonic-net/sonic-buildimage/pull/15569

**Comment by liuh-80** [comment]
@nazariig , this PR has merge conflict when on 202012 branch, please create manually cherry-pick PR to 202012.

**Comment by nazariig** [comment]
> @nazariig , this PR has merge conflict when on 202012 branch, please create manually cherry-pick PR to 202012.

@liuh-80 done

**Pull request merged**